guest/api/types.go

```diff
@@ -122,6 +126,13 @@ type ReservePlugin interface {
 	Unreserve(state CycleState, p proto.Pod, nodeName string)
 }
 
+// PermitPlugin is a WebAssembly implementation of framework.PermitPlugin.
+type PermitPlugin interface {
+	Plugin
+
+	Permit(state CycleState, p proto.Pod, nodeName string) (*Status, time.Duration)
```

Comment:
Document `time.Duration` should be in the range of uint32 and otherwise the guest panic (?).

Reply:
We need to constrain the range of `time.Duration` as you mentioned.
So the host call `set_timeout` that I first proposed in this PR is no longer needed, right?

Reply:
I thought current implementation is fine because timeout is being passed via linear memory instead of stack. Only status is being passed via stack. So we don't need to restrict the value of timeout in uint32 range. Do I misunderstand?

Reply:
> So we don't need to restrict the value of timeout in uint32 range. Do I misunderstand?

We do. `Int64ToPtr` wouldn't behave as user's intention when timeout value is too big. (see: https://go.dev/play/p/tDCFt2Yu0Td)

> So the host call set_timeout that I first proposed in this PR is no longer needed, right?
I'd like to pass the statusCode and time.Duration to host by packing them into a single i64 return value.

LGTM.